crates/energy/benches/nearby.rs

```diff
@@ -0,0 +1,152 @@
+use bevy::ecs::schedule::ScheduleLabel;
+use bevy::prelude::{
+    App, Changed, Component, Entity, Query, Res, ResMut, Resource, Schedule, Transform, Update,
+    World,
+};
+use bevy::time::TimePlugin;
+use criterion::{criterion_group, criterion_main, Criterion};
+use de_core::projection::ToAltitude;
+use de_energy::{update_nearby_recv, EnergyReceiver, NearbyUnits};
+use de_index::{EntityIndex, LocalCollider};
+use de_objects::ObjectCollider;
+use de_test_utils::load_points;
+use parry3d::math::{Isometry, Vector};
+use parry3d::shape::{Cuboid, TriMesh};
+
+const MAP_SIZE: f32 = 2000.;
+const DISTANCE_FROM_MAP_EDGE: f32 = 100.;
+const SPEED: f32 = 10.; // based on MAX_H_SPEED in movement
+
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct UpdateUnits;
+
+#[derive(Component)]
+struct UnitNumber(u32);
+
+#[derive(Component)]
+struct Centre {
+    x: f32,
+    y: f32,
+}
+
+#[derive(Resource)]
+struct Clock(f32); // this clock is used in a substitute of time to make it more deterministic
+
+impl Clock {
+    fn inc(&mut self) {
+        self.0 += 0.008 // 125 updates a "second"
+    }
+}
+
+fn update_index(
+    mut index: ResMut<EntityIndex>,
+    moved: Query<(Entity, &Transform), Changed<Transform>>,
+) {
+    for (entity, transform) in moved.iter() {
+        let position = Isometry::new(
+            transform.translation.into(),
+            transform.rotation.to_scaled_axis().into(),
+        );
+        index.update(entity, position);
+    }
+}
+
+fn init_world_with_entities_moving(world: &mut World, num_entities: u32) {
+    let mut index = EntityIndex::new();
+    let points = load_points(
+        num_entities.try_into().unwrap(),
+        MAP_SIZE - DISTANCE_FROM_MAP_EDGE,
```

Comment:
We should multiply by two here, since there are two margins (one on each side). This is mathematically corrected by the division by two elsewhere (see my comment there) but semantically, the correction belongs here.

Also, I would store the result to a variable and assert that its greater than 0.